src/main.py

```diff
@@ -0,0 +1,56 @@
+# OMIT !/usr/bin/env python3
+from pathlib import Path
+import click
+from io import StringIO
+
+from src.option_select import select
+from src.run_cocci import run_cocci
+
+@click.command
+@click.option(
+    "-i",
+    "--input",
+    type=Path,
+    required=True,
+    help="Directory containing your C source files.",
+)
+@click.option(
```

Comment:
Your implementation of this looks fun from the developer's perspective, but from the usability standpoint, it's easier to be able to simply specify which patch you want to run directly when calling this command. On windows I use  `Cmder`, a console emulator, so I can use the `ctrl + R` shortcut to run previously executed commands. That makes it easy to e.g. run one patch against multiple files, or after making some fixes. You don't have to pick a patch every time you want to run just that one. By default, if not patch is specified, run all